src/invidious/metrics.cr

```diff
@@ -0,0 +1,48 @@
+# Module containing an optional Kemal handler, which can be used
+# to collect metrics about the usage of various Invidious routes.
+module Metrics
+  record MetricLabels, request_method : String, request_route : String, response_code : Int32
+
+  # Counts how many a given route was used
+  REQUEST_COUNTERS              = Hash(MetricLabels, Int64).new
+
+  # Counts how much time was used to handle requests to each route
+  REQUEST_DURATION_SECONDS_SUMS = Hash(MetricLabels, Float32).new
+
+  # The handler which will record metrics when registered in a Kemal application 
+  METRICS_COLLECTOR = RouteMetricsCollector.new(REQUEST_COUNTERS, REQUEST_DURATION_SECONDS_SUMS)
```

Comment:
Please initialize a new `RouteMetricsCollector` at the same place that you declare a new handler (in invidious.cr, see my other comment). In general, it's better to avoid singletons (particularly in this case, where this "constant" is used only once).

Reply:
Sure, it's initialised along with the handler registration code now